src/parameters.jl

```diff
@@ -0,0 +1,603 @@
+# Copyright (c) 2020: Akshay Sharma and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+# block other methods
+
+MOI.supports(::POI.Optimizer, ::ForwardObjectiveFunction) = false
+
+function MOI.set(::POI.Optimizer, ::ForwardObjectiveFunction, _)
+    return error(
+        "Forward objective function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to set the forward sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ForwardConstraintFunction) = false
+
+function MOI.set(
+    ::POI.Optimizer,
+    ::ForwardConstraintFunction,
+    ::MOI.ConstraintIndex,
+    _,
+)
+    return error(
+        "Forward constraint function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to set the forward sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ReverseObjectiveFunction) = false
+
+function MOI.get(::POI.Optimizer, ::ReverseObjectiveFunction)
+    return error(
+        "Reverse objective function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to get the reverse sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ReverseConstraintFunction) = false
+
+function MOI.get(
+    ::POI.Optimizer,
+    ::ReverseConstraintFunction,
+    ::MOI.ConstraintIndex,
+)
+    return error(
+        "Reverse constraint function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to get the reverse sensitivity.",
+    )
+end
+
+# functions to be used with ParametricOptInterface.jl
+
+struct ForwardConstraintSet <: MOI.AbstractConstraintAttribute end
+
+struct ReverseConstraintSet <: MOI.AbstractConstraintAttribute end
+
+mutable struct SensitivityData{T}
+    parameter_input_forward::Dict{MOI.VariableIndex,T}
+    parameter_output_backward::Dict{MOI.VariableIndex,T}
+end
+
+function SensitivityData{T}() where {T}
+    return SensitivityData{T}(
+        Dict{MOI.VariableIndex,T}(),
+        Dict{MOI.VariableIndex,T}(),
+    )
+end
+
+function _get_sensitivity_data(
+    model::POI.Optimizer{T},
+)::SensitivityData{T} where {T}
+    _initialize_sensitivity_data!(model)
+    return model.ext[:_sensitivity_data]::SensitivityData{T}
+end
+
+function _initialize_sensitivity_data!(model::POI.Optimizer{T}) where {T}
+    if !haskey(model.ext, :_sensitivity_data)
```

Comment:
Use a constant, I don't like copy-pasting symbols because a typo does not trigger a `UndefVarError` and so it's harder to debug ^^